app/jobs/search_keyword_job.rb

```diff
@@ -5,22 +5,38 @@ class SearchKeywordJob < ApplicationJob
 
   def perform(search_result_id)
     search_result = SearchResult.find(search_result_id)
-    html_code = Search::SearchService.new(
-      keyword: search_result.keyword,
-      search_engine: search_result.search_engine
-    ).search
+    html_code = search(search_result.keyword, search_result.search_engine)
 
-    return mark_as_failed(search_result) unless html_code
+    result = parse(search_result.search_engine, html_code)
 
-    mark_as_completed(search_result, html_code)
-  rescue IcRailsHueyToby::Errors::SearchServiceError
+    set_parsing_result(search_result, html_code, result)
+  rescue IcRailsHueyToby::Errors::SearchServiceError, IcRailsHueyToby::Errors::ParseServiceError
     mark_as_failed(search_result)
   end
 
   private
 
-  def mark_as_completed(search_result, html_code)
-    search_result.update html_code: html_code, status: :completed
+  def search(keyword, search_engine)
+    Search::SearchService.new(
+      keyword: keyword,
+      search_engine: search_engine
+    ).search
+  end
+
+  def parse(search_engine, html_code)
+    Parse::ParseService.new(
+      search_engine: search_engine,
+      html_code: html_code
+    ).parse
+  end
+
+  def set_parsing_result(search_result, html_code, result)
+    search_result.update result.merge({ html_code: html_code })
```

Comment:
We could potentially do `.merge({ html_code: html_code })` inside `parse_service`, since we pass it as a parameter already anyways, thoughts? 💭

Reply:
@Tuubz By merging the `html_code` inside google_parse_service, we will need to update the test case to expect the html_code from returning result. Because the html_code is quite long so we should skip that testcase. What do you think?